reckoner/chart.py

```diff
@@ -191,63 +185,162 @@ def install(self, namespace=None, context=None):
-                self.args.append("--set={}={}".format(k, v))
 
+        # HACK: This is in place until we can fully deprecate the usage of set
+        #       in place of values. Currently values: gets translated into
+        #       --set commands to the helm command line run. This should be
+        #       changed so that set: goes to command line args and values: goes
+        #       to temporary values files, to keep strong types in yaml.
+        #       This will go away once we fully change the functionality of
+        #       values: vs sets: settings in course.yml
+        self._merge_set_and_values()
+
+        # Build the list of existing yaml files to use for the helm command
+        self.build_files_list()
+
+        # Build the file arguments from the `values: {}` in course.yml
+        self.build_temp_values_files()
+
+        # Build the list of --set arguments
+        self.build_set_arguments()
+
+        # Build the list of --set-string arguments
+        self.build_set_string_arguments()
+
+    def _merge_set_and_values(self):
+        '''
+        This is a temporary method that will be gone once the values: vs set:
+        debacle has been resolved. (See https://github.com/reactiveops/reckoner/issues/7)
+
+        NOTE ONLY RUN THIS ONCE BECAUSE IT'S NOT IDEMPOTENT
+        '''
+        if self.__already_merged:
+            raise StandardError('This method cannot be called twice. '
+                                'If you are seeing this please open an '
+                                'issue in github.')
+
+        def merge_dicts(values, sets):
+            '''This does a dict merge and prefers "sets" values'''
+            new_dict = values.copy()
+            new_dict.update(sets)
+            return new_dict
+
+        logging.debug('Merging values: into sets: - sets: take precedence.')
+        logging.debug('Original value of sets: {}'.format(self.sets))
+        self.sets = merge_dicts(self.values, self.sets)
+        logging.debug('New value of sets: {}'.format(self.sets))
+        self.__already_merged = True
+
+    def build_temp_values_files(self):
+        '''
+        This method builds temporary files based on the values: settings
+        provided in the course.yml. This effectively keeps type persistence
+        between the course.yml and what is passed to helm. If you use set:
+        value arguments then you can lose types like int, float and true/false
+        '''
+        logging.warning(
+            "DEPRECATION NOTICE: The behavior of `values: {}` will change in "
+            "v0.14+. Currently values: are set as --set arguments for the "
+            "helm command. This behavior makes all values: you set in the "
+            "course.yml lose type fidelity, true becomes string(\"true\") "
+            "and all int values become a string of the integer. See "
+            "https://github.com/reactiveops/reckoner/issues/7 for details."
+        )
```

Comment:
This should have more instructional content. something that says change your values key two sets to keep the same behavior.